packages/nimble-components/src/dialog/specs/README.md

```diff
@@ -157,7 +198,7 @@ By using the native `dialog` element, we get good a11y behavior without having t
 -   ESC key closes the dialog
 -   The native dialog defaults to the a11y role `dialog` (but we will override this, as discussed below)
 
-The role `alertdialog` should be used for most modal dialogs (since they demand the user's attention), so we will set it by default. When role is `alertdialog`, `aria-describedby` is also supposed to be set to reference the element containing the alert message (which would be part of the user-provided content). Unfortunately, because the `dialog` and the alert message element are on different sides of the shadow DOM boundary, we can't reference the element ID. Similarly, `aria-labelledby` is supposed to point to an element that is the title for the dialog, but it has the same problem with referencing IDs. Instead, `aria-label` should provide the title directly. It is up to the client to set `aria-label` appropriately, and we will synchronize it between `nimble-dialog` and the `dialog` element in the shadow root.
+The role `alertdialog` should be used for most modal dialogs (since they demand the user's attention), so we will set it by default. When role is `alertdialog`, `aria-describedby` should be set to reference the element containing the alert message. Therefore, `aria-describedby` will reference the `content` provided by the dialog. Similarly, `aria-labelledby` should point to an element that is the title for the dialog, so it will reference the `title` slot of the dialog. Therefore, a client should always provide an appropriate title for their dialog, even if `header-hidden` is `true`.
```

Comment:
I did a bit of reading and experimentation, and here are my findings:
- Based on W3C [here](https://w3c.github.io/aria/#alertdialog) and [here](https://www.w3.org/WAI/ARIA/apg/patterns/alertdialog/), an alertdialog should have a description set. I don't see anything in the W3C docs about using `aria-details` instead.
- It doesn't seem like setting `aria-describedby` to complex content is a good idea because the accessibility tree combines all the text together with no indication of the context. For example, the description for the delete test result confirmation would be "Are you sure you want to permanently delete the selected result? Delete all attachments associated with the selected result" with no indication that the second sentence was associated with a checkbox.
- `aria-describedby` is optional for the `dialog` role. Specifically, the [docs](https://www.w3.org/WAI/ARIA/apg/patterns/dialogmodal/) say:
    > It is advisable to omit specifying aria-describedby if the dialog content includes semantic structures, such as lists, tables, or multiple paragraphs, that need to be perceived in order to easily understand the content, i.e., if the content would be difficult to understand when announced as a single unbroken string.

Based on this, I lean towards changing the default role of the `nimble-dialog` to be `dialog` rather than `alertdialog`. I think for something as generic as we are trying to create in nimble, there are too many assumptions (such as the content only being text) associated with the `alertdialog`. Then, we could automatically set the `aria-label` based on the title/subtitle, but we would not set the `aria-description`.

Any thoughts or objections to changing the role of the `nimble-dialog` to `dialog`? @m-akinc @rajsite

Reply:
I think your evaluation makes sense that using the `dialog` role is acceptable since they are allowed to be modal and have complex content. Maybe a future enhancement could be making it easier to configure it as `alertdialog` but a default of `dialog` today seems fine.

Reply:
I've updated the spec. We'll use a role of `dialog` with `modal: true` (we get both of those for free with the `dialog` element), and we'll only set `aria-labelledby`. We will not do anything with `aria-describedby`.